src/prototype_creep_harvest.js

```diff
@@ -43,8 +43,28 @@ Creep.prototype.handleSourcer = function() {
 };
 
 Creep.prototype.spawnCarry = function() {
-  var energyThreshold = Game.rooms[this.memory.base].controller.level * config.sourcer.spawnCarryLevelMultiplier;
-  var waitTime = config.sourcer.spawnCarryWaitTime;
+  if (this.memory.wait > 0) {
+    this.memory.wait -= 1;
+    return false;
+  }
+
+  var foundKey;
+  for (let key of Object.keys(config.carry.sizes).sort(function(a, b) { return parseInt(a, 10) - parseInt(b, 10); })) {
```

Comment:
regarding converting I ran exactly into the problem that it was wrong. Had the same issue with different browser today.

Possible that the behavior changed recently.

I'm not an nodejs expert. Is the sorting executed for each element again or only once. I can move it to a variable before?!